builtin/providers/aws/resource_aws_security_group_rule.go

```diff
@@ -288,6 +308,23 @@ func ipPermissionIDHash(ruleType string, ip *ec2.IPPermission) string {
 		}
 	}
 
+	if len(ip.UserIDGroupPairs) > 0 {
+		sort.Sort(byUserIDAndGroup(ip.UserIDGroupPairs))
+		for _, pair := range ip.UserIDGroupPairs {
+			buf.WriteString(fmt.Sprintf("%s-", *pair.UserID))
```

Comment:
I think this needs to be removed; it's causing TF to always want to re-make the rule because we don't create (or store) the `UserID` when we create the rule. The `perm` we send does not have this populated, which I think is OK since the documentation say it's for EC2 Classic only anyway. 

I commented it out and this branch started to behave as expected.